vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc

```diff
@@ -0,0 +1,224 @@
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+std::vector<PaymentInfo> Payments::GetPaymentsFromList(
+    base::ListValue* list) const {
+  DCHECK(list);
+
+  std::vector<PaymentInfo> payments;
+
+  for (auto& item : *list) {
+    base::DictionaryValue* dictionary;
+    if (!item.GetAsDictionary(&dictionary)) {
+      continue;
+    }
+
+    PaymentInfo payment;
+
+    if (!GetBalanceFromDictionary(
+        dictionary, &payment.balance)) {
+      continue;
+    }
+
+    if (!GetTransactionCountFromDictionary(
+        dictionary, &payment.transaction_count)) {
+      continue;
+    }
+
+    payments.push_back(payment);
+  }
+
+  return payments;
+}
+
+bool Payments::GetBalanceFromDictionary(
+    base::DictionaryValue* dictionary,
+    double *balance) const {
+  DCHECK(dictionary);
+  DCHECK(balance);
+
+  auto* balance_value = dictionary->FindKey("balance");
+  if (!balance_value) {
+    return false;
+  }
+
+  *balance = std::stod(balance_value->GetString());
+
+  return true;
+}
+
+bool Payments::GetTransactionCountFromDictionary(
+    base::DictionaryValue* dictionary,
+    uint64_t *transaction_count) const {
+  DCHECK(dictionary);
+  DCHECK(transaction_count);
+
+  auto* transaction_count_value = dictionary->FindKey("transactionCount");
+  if (!transaction_count_value) {
+    return false;
+  }
+
+  *transaction_count = std::stoull(transaction_count_value->GetString());
+
+  return true;
+}
+
+int Payments::CalculateNexyPaymentMonth(
+    const base::Time::Exploded& exploded,
+    const uint64_t next_token_redemption_date_in_seconds) const {
+  int month = exploded.month;
+
+  if (exploded.day_of_month <= kNextPaymentDay) {
+    if (HasPendingBalanceForLastMonth()) {
+      // If last month has a pending balance, then the next payment date will
+      // occur this month
+    } else {
+      // If last month does not have a pending balance, then the next payment
+      // date will occur next month
+      month++;
+    }
+  } else {
+    if (HasPendingBalanceForThisMonth()) {
+      // If this month has a pending balance, then the next payment date will
+      // occur next month
+      month++;
+    } else {
+      auto next_token_redemption_date =
+          base::Time::FromDoubleT(next_token_redemption_date_in_seconds);
+
+      base::Time::Exploded next_token_redemption_date_exploded;
+      next_token_redemption_date.LocalExplode(
+          &next_token_redemption_date_exploded);
+
+      if (next_token_redemption_date_exploded.month == month) {
+        // If this month does not have a pending balance and our next token
+        // redemption date is this month, then the next payment date will occur
+        // next month
+        month++;
+      } else {
+        // If this month does not have a pending balance and our next token
+        // redemption date is next month, then the next payment date will occur
+        // the month after next
+        month += 2;
```

Comment:
Can this function return a month greater than 12? If so, maybe we should note that somewhere in a comment (or have it return the correct month along with a year or boolean flag)?